vendor.sh

```diff
@@ -46,15 +65,15 @@ function download_typescript() {
     files="parser.c scanner.c"
 
     echo "downloading typescript $version"
-    mkdir -p "typescript/common"
-    curl -s -f -S "https://raw.githubusercontent.com/tree-sitter/tree-sitter-typescript/$version/common/scanner.h" -o "typescript/common/scanner.h"
     for lang in $langs; do
-        mkdir -p "typescript/$lang/tree_sitter"
-        curl -s -f -S "https://raw.githubusercontent.com/tree-sitter/tree-sitter-typescript/$version/$lang/src/tree_sitter/parser.h" -o "typescript/$lang/tree_sitter/parser.h"
+        curl -s -f -S "https://raw.githubusercontent.com/tree-sitter/tree-sitter-typescript/$version/common/scanner.h" -o "typescript/$lang/scanner.h"
+        curl -s -f -S "https://raw.githubusercontent.com/tree-sitter/tree-sitter-typescript/$version/$lang/src/tree_sitter/parser.h" -o "typescript/$lang/parser.h"
         for file in $files; do
             curl -s -f -S "https://raw.githubusercontent.com/tree-sitter/tree-sitter-typescript/$version/$lang/src/$file" -o "typescript/$lang/$file"
-            sed -i '' -e 's/\.\.\/\.\./\.\./g' "typescript/$lang/$file"
+            sed -i 's/"\.\.\/\.\.\/common\/scanner\.h"/"scanner\.h"/g' "typescript/$lang/$file"
```

Comment:
this change breaks the script on macOs.

Reply:
And the version with `-i '' -e '...'` doesn't work on Linux. Rewriting this script in Go may be way more portable. WDYT?

Reply:
answered in another pr